### PR TITLE
allow @ in resource part

### PR DESCRIPTION
**Pull request by sonnyp**

### CODE REVIEW

File: lib/JID.js [hunk 1]
```diff
@@ -25,14 +25,18 @@ function JID (a, b, c) {
...
-  if (s.indexOf('/') >= 0) {
-    this.setResource(s.substr(s.indexOf('/') + 1))
-    s = s.substr(0, s.indexOf('/'))
+
+  var atStart = s.indexOf('@')
+  if (atStart !== -1) {
+    this.setLocal(s.substr(0, atStart))
+    s = s.substr(atStart + 1)
   }
+
```
Comment (sonnyp) [inline lib/JID.js]: @lloydwatkin 

https://tools.ietf.org/html/rfc7622#section-3.2 states that the resource should be parsed first.
>  This parsing order is important, as illustrated by example 15 in
   Section 3.5.

This fixes the issue with @ in resource part which is a common use case but it breaks other tests related to JID escaping (XEP 0106 http://xmpp.org/extensions/xep-0106.html)

I think we're trying to be too smart about JID escaping please have a look at how I fixed the tests below. If you agree I'll apply the same modification to all JID escaping tests + add a note to the README about how users should always use ```new JID(node, domain, resource)``` if they care about JID escaping and why.

Comment (lloydwatkin) [inline lib/JID.js]: If we're breaking tests but are more compliant then I don't think there's an argument against the changes.

Comment (sonnyp) [inline lib/JID.js]: @lloydwatkin agree, applying the same change to all test + adding a README doc and a new release

**Comment by sonnyp** [comment]
@lloydwatkin can you review this please?

**Comment by lloydwatkin** [comment]
Late night question, is it possible to have an escaped slash in the local part?

Otherwise all good!

**Comment by sonnyp** [comment]
@lloydwatkin according to https://tools.ietf.org/html/rfc7622#section-3.4
`/` doesn't need to be escaped in the resource part

> for example, an XMPP
      address of the form <localpart@domainpart/foo/bar> does not
      identify a resource "bar" that exists below a resource "foo" in a
      hierarchy of resources associated with the entity

**Comment by sonnyp** [comment]
@lloydwatkin I just added a test for slash in resource part https://github.com/node-xmpp/JID/commit/d0b177fd6a88219d1aa3d4e8575f0848f46384e2

**Pull request merged**